packages/stencil/components/menu/src/components/osds-menu/osds-menu.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component, Prop, h, Host, Watch, Element, Listen, State } from '@stencil/core';
+import {
+  OdsMenu,
+  OdsMenuController,
+  OdsMenuEvents,
+  OdsMenuItem,
+  OdsMenuMethods,
+  odsMenuDefaultAttributes,
+} from '@ovhcloud/ods-core';
+import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
+import { ocdkDefineCustomElements, ocdkIsSurface, OcdkSurface, } from '@ovhcloud/ods-cdk';
+import { HTMLStencilElement } from '@stencil/core/internal';
+
+ocdkDefineCustomElements();
+
+@Component({
+  tag: 'osds-menu',
+  styleUrl: 'osds-menu.scss',
+  shadow: true,
+})
+export class OsdsMenu implements OdsMenu<OdsStencilMethods<OdsMenuMethods>, OdsStencilEvents<OdsMenuEvents>> {
+  controller: OdsMenuController = new OdsMenuController(this as OdsMenu);
+  title: HTMLElement | null = null;
+  anchor!: HTMLDivElement;
+  surface: OcdkSurface | undefined = undefined;
+
+  @Element() el!: HTMLStencilElement;
+
+  @Prop({ reflect: true }) public disabled: boolean | undefined = odsMenuDefaultAttributes.disabled;
+
+  @State() tabindex = 0;
+
+  @Watch('disabled')
+  propagateDisabledToChild(disabled: boolean | undefined): void {
+    this.controller.propagateDisabledToChild(disabled);
```

Comment:
maybe it would be easier to just watch the disabled prop, close the surface when true and avoid all this propagation stuff ?